app/modules/users/resources.py

```diff
@@ -107,19 +106,19 @@ class UserByID(Resource):
     Manipulations with a specific user.
     """
 
-    @api.permission_required(
-        permissions.ObjectAccessPermission,
-        kwargs_on_request=lambda kwargs: {
-            'obj': kwargs['user'],
-            'action': AccessOperation.READ,
-        },
-    )
-    @api.response(schemas.DetailedUserSchema())
+    # no decorators here for permissions or schemas, as getting specific user data is "public"
+    #   with the schema being determined based on privileges below
     def get(self, user):
         """
         Get user details by ID.
         """
-        return user
+        from app.modules.users.permissions import rules
+        from app.modules.users.schemas import DetailedUserSchema, PublicUserSchema
+        if not current_user.is_anonymous and (rules.owner_or_privileged(current_user, user) or current_user.is_admin):
+            schema = DetailedUserSchema()
+        else:
+            schema = PublicUserSchema()
```

Comment:
This does seem inconsistent in terms of permission. Only admin users can read the list of user GUIDs, but anyone can read their names once they have the guid.
I may have missed a conversation but this seems very strange

Reply:
after some slack discussion, i think indeed there will be some addressing we need to do here -- especially making sure what is exposed is consistent with our terms & conditions etc.  basically this is a kind of open area of discussion.